### PR TITLE
Adds drag/drop image uploading function to comment editor

### LINKED ISSUE #17882: Add drag/drop image uploading function in comment editor
📖 As a commenter, when drafting/editing a comment, I can drag & drop an image into a comment I am editing (i.e. where the cursor is focused). The dropped image should be uploaded and inserted for me into the comment without me having to manually format further. 

Some additional context which might be helpful:

- We achieve this functionality in the editor by using the `useDragAndDrop` Preact hook
- Comment text areas are _not_ initially rendered in Preact, but once focused, they are replaced by the Preact `MentionAutocompleteTextArea` using `window.Forem.initializeMentionAutocompleteTextArea`

### PR DESCRIPTION
**Pull request by rt4914**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Adds drag/drop image uploading function to comment editor.
As a commenter, when drafting/editing a comment, I can drag & drop an image into a comment I am editing (i.e. where the cursor is focused). The dropped image should be uploaded and inserted for me into the comment without me having to manually format further.

## Related Tickets & Documents

- Related Issue #17882
- Closes #17882

## QA Instructions, Screenshots, Recordings

- Visit any post and under comment section simply drag-and-drop image to see that it uploads correctly.

https://user-images.githubusercontent.com/9396084/183449789-3f9014c9-fefe-44b2-84a1-a9c70873d512.mov

### UI accessibility concerns?

No

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: because here we are using files/code which has already been tested in https://github.com/forem/forem/pull/10145/files
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [x] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

No

## [optional] What gif best describes this PR or how it makes you feel?

Copy/Paste

### CODE REVIEW

File: app/javascript/packs/CommentTextArea/CommentTextArea.jsx [hunk 1]
```diff
@@ -36,13 +42,33 @@ const handleImageSuccess = (textAreaRef) => {
     editableBodyElement.selectionStart =
       selectionStart + markdownImageLink.length;
     editableBodyElement.selectionEnd = editableBodyElement.selectionStart;
+
+    // Dispatching a new event so that linkstate, https://github.com/developit/linkstate,
+    // the function used to create the onChange prop gets called correctly.
+    editableBodyElement.dispatchEvent(new Event('input'));
```
Comment (aitchiss) [inline app/javascript/packs/CommentTextArea/CommentTextArea.jsx:48]: The comment area doesn't use `linkstate` so I think it should be safe to remove this? I tried locally and it seems to work as expected, but maybe I'm missing something!

**Review by aitchiss — APPROVED** [review]
Works great! ✨

File: app/javascript/packs/CommentTextArea/CommentTextArea.jsx (hunk 1 above)
Comment (rt4914) [reply app/javascript/packs/CommentTextArea/CommentTextArea.jsx]: Removed as it is not required.

**Review by Ridhwana — APPROVED** [review]
Looks and works great @rt4914 ✨

**Pull request merged**